src/app/modules/submission/encrypt-submission/encrypt-submission.utils.ts

```diff
@@ -237,3 +240,26 @@ export const createEncryptedSubmissionDto = (
     version: submissionData.version,
   }
 }
+
+export const getPaymentAmount = (
+  formPaymentFields: FormPaymentsField, // fields that are from document.form
+  incomingSubmissionPaymentFields?: PaymentFieldsDto, // fields that are from incoming submission
+) => {
+  if (!formPaymentFields) {
+    return
+  }
+  // legacy payment forms may not have a payment type
+  const { payment_type } = formPaymentFields
+  switch (payment_type) {
+    case PaymentType.Fixed:
+      return formPaymentFields.amount_cents
+    case PaymentType.Variable:
+      return incomingSubmissionPaymentFields?.amount_cents
+
+    default: {
+      // Force TS to emit an error if the cases above are not exhaustive
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const exhaustiveCheck: never = payment_type
```

Comment:
why not return empty like in line 249? what is the implication of forcing an error here?

Reply:
Typescript would explicitly fail to compile if we missed out a case. By returning `empty` it would mean that the code can still compile, and server would fail during runtime.

Reply:
how would server fail it though? abruptly with exit(1)? or by throwing error? I think we can do graceful exit in this case?

Reply:
As in the behaviour ~would~could be undefined (or rather don't really want to think about it). If we could detect that such potential errors could exist, we can should try to handle them. Similar to most statically-typed languages forces you to account for all cases before allowing one to emit an executable.

Reply:
got it, thanks!